src/built_ins/cd.c

```diff
@@ -61,8 +61,7 @@ void	call_cd(t_data *data, char **cmd)
 	char	*p;
 	char	*cwd;
 
-	cwd = NULL;
-	if (cmd[1] == NULL || ft_strcmp(cmd[1], "-") == 0)
+	if (!cmd[1] || cmd[1][0] == '-')
```

Comment:
Use of logicOR here is incorrect as it will not short-circuit correctly.

The second part of the statement will cause a segfault if cmd[1] points to NULL, whereas ft_strcmp is protected against it:
![image](https://github.com/user-attachments/assets/aec00064-88ce-4639-9e1f-0455e2fceee7)

Reply:
[1a5e0a4](https://github.com/Haliris/minishell/pull/48/commits/1a5e0a409686cb383022078e30e06931feff7e3a)